### PR TITLE
(FACT-1144) Add memory resolver for OpenBSD

### PR DESCRIPTION
**Pull request by jasperla**
Currently without `swap` facts while those are being sorted out.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Mind commenting with the output from the resolver?

**Comment by jasperla** [comment]
```
{
  system => {
    available => "1.47 GiB",
    available_bytes => 1578127360,
    capacity => "59.23%",
    total => "3.61 GiB",
    total_bytes => 3870953472,
    used => "2.14 GiB",
    used_bytes => 2292826112
  }
}
```

File: lib/src/facts/openbsd/memory_resolver.cc [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        } else {
+                int uvmexp_mib[] = { CTL_VM, VM_UVMEXP };
+                struct uvmexp uvmexp;
+                len = sizeof(uvmexp);
+                if (sysctl(uvmexp_mib, 2, &uvmexp, &len, nullptr, 0) == -1) {
+                    LOG_DEBUG("sysctl uvmexp failed: %1% (%2%): free memory is not available.", strerror(errno), errno);
+                }
+
+                // Should we account for the buffer cache?
+                result.mem_total = (uint64_t) uvmexp.npages << uvmexp.pageshift;
```
Comment (peterhuene) [inline lib/src/facts/openbsd/memory_resolver.cc]: We generally prefer c++-style casts over c-style casts (for various reasons).  `result.mem_total = static_cast<uint64_t>(uvmexp.npages << uvmexp.pageshift);` should do it.  Same comment for the line below.

File: lib/src/facts/openbsd/memory_resolver.cc [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+using namespace std;
+using namespace facter::execution;
+using namespace facter::util;
+
+namespace facter { namespace facts { namespace openbsd {
+
+    memory_resolver::data memory_resolver::collect_data(collection& facts)
+    {
+        data result;
+        size_t len;
```
Comment (peterhuene) [inline lib/src/facts/openbsd/memory_resolver.cc]: Can we move this declaration to its assignment?  I prefer not to declare variables apart from where they are initialized unless I have to.

**Comment by peterhuene** [comment]
Other than above minor comments, this looks good :+1:

**Comment by jasperla** [comment]
I've fixed up the casts and declaration.

File: lib/src/facts/openbsd/memory_resolver.cc [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        data result;
+
+        // Get the system page size
+        int pagesize_mib[] = { CTL_HW, HW_PAGESIZE};
+        int page_size = 0;
+        size_t len = sizeof(page_size);
+        if (sysctl(pagesize_mib, 2, &page_size, &len, nullptr, 0) == -1) {
+            LOG_DEBUG("sysctl failed: %1% (%2%): system page size is unknown.", strerror(errno), errno);
+        } else {
+                int uvmexp_mib[] = { CTL_VM, VM_UVMEXP };
```
Comment (peterhuene) [inline lib/src/facts/openbsd/memory_resolver.cc]: Looks like this block is unintentionally indented more than needed.

**Comment by jasperla** [comment]
That's also fixed now.

**Comment by HAIL9000** [comment]
@peterhuene, could you take another look now that it's been updated?

**Comment by peterhuene** [comment]
@jasperla Thanks again for the contribution!

**Pull request merged**